sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js

```diff
@@ -0,0 +1,113 @@
+/** @odoo-module **/
+/* Copyright 2019 Tecnativa - Ernesto Tejeda
+/* Copyright 2022 Tecnativa - Víctor Martínez
+/* Copyright 2023 Tecnativa - Yadier Quesada
+/* License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).*/
+
+const {useState} = owl;
+import {BooleanField} from "@web/views/fields/boolean/boolean_field";
+import {_lt} from "@web/core/l10n/translation";
+import {registry} from "@web/core/registry";
+
+const iconTrue = "fa-check-square-o";
+const iconFalse = "fa-square-o";
+const tooltipTrue = "Switch to hidden";
+const tooltipFalse = "Switch to show";
+
+export class BooleanFaIconWidget extends BooleanField {
+    setup() {
+        super.setup();
+        this.state = useState({
+            fa_class: this.getFaIconClass(this.props.value),
+            text_tooltip: this.getTextTooltip(this.props.value),
+        });
+    }
+
+    getFaIconClass(currentValue) {
+        var fa_icons = this.props.fa_icons;
+        var icon_true = fa_icons.icon_true || iconTrue;
+        var icon_false = fa_icons.icon_false || iconFalse;
+        return currentValue ? icon_true : icon_false;
+    }
+
+    getTextTooltip(currentValue) {
```

Comment:
Why not using a getter to simplify everythin related with this? Take a look at this reference: https://github.com/odoo/odoo/blob/ce3e66609e70cef6801baf9dba71c84798d382c1/addons/web/static/src/views/fields/field.js#L186

Reply:
Nice example, but it doesn't work at the moment, I tried before and the behavior was not as expected.
This property is as if it were computed, it depends on the onchange event of this.props.value.
I even used an asynchronous event, but the fa_icons and text_tooltip state always fails the first time it's clicked.
If you can take a look and propose a better solution, you're welcome.